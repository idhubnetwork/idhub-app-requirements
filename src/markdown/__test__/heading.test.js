import Markdown from 'markdown-it'
import headingsPlugin from '../headings'

test('main', () => {
  const md = new Markdown()
  md.use(headingsPlugin)

  const env = {
    headings: []
  }
  const html = md.render(
    `# foo

## hehe

### well good

## wow

## wow

## wow
  `,
    env
  )

  expect(html).toMatchSnapshot('html')
  expect(env).toMatchSnapshot('env')
})
