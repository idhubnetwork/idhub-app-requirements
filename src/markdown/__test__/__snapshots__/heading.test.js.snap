// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main: env 1`] = `
Object {
  "headings": Array [
    Object {
      "depth": 2,
      "id": "hehe",
      "value": "hehe",
    },
    Object {
      "depth": 3,
      "id": "well-good",
      "value": "well good",
    },
    Object {
      "depth": 2,
      "id": "wow",
      "value": "wow",
    },
    Object {
      "depth": 2,
      "id": "wow1",
      "value": "wow",
    },
    Object {
      "depth": 2,
      "id": "wow2",
      "value": "wow",
    },
  ],
  "ids": Array [
    "foo",
    "hehe",
    "well-good",
    "wow",
    "wow",
    "wow",
  ],
  "title": "foo",
}
`;

exports[`main: html 1`] = `
"
<h2 id=\\"hehe\\" tabindex=\\"0\\">hehe</h2>
<h3 id=\\"well-good\\" tabindex=\\"0\\">well good</h3>
<h2 id=\\"wow\\" tabindex=\\"0\\">wow</h2>
<h2 id=\\"wow1\\" tabindex=\\"0\\">wow</h2>
<h2 id=\\"wow2\\" tabindex=\\"0\\">wow</h2>
"
`;
